### PR TITLE
Chapter3/Homework/7And8

### PR DESCRIPTION
**Pull request by matthew-hole**
When creating the custom Linq Extensions for Shuffle, I attempted to make it as generic as I could by accepting any IEnumerable<T>.

This homework was quite fun but also challenging and allowed me to learn more about the inner workings of the language. As always, please point out anyways I can improve - whether its structure of the code of simplification of classes/methods I have created.

Once again, thanks for the great lessons and homeworks!

### CODE REVIEW

File: Src/BootCamp.Chapter/CollectionWriter.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace BootCamp.Chapter
+{
+    public static class CollectionWriter
+    {
+        public static void WriteCollectionToConsole<T>(IEnumerable<T> collection)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CollectionWriter.cs:8]: Simplify name: Print()

File: Src/BootCamp.Chapter/LINQExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public static class LINQExtensions
+    {
+        /// <summary>
+        /// Generic Shuffling Algorithm which will accept a collection of type T and return a new shuffled List<T>.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="collection"></param>
+        /// <returns></returns>
+        public static IEnumerable<T> Shuffle<T>(this IEnumerable<T> collection)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/LINQExtensions.cs:15]: This works, a bit inneficient though. A more efficient way would be to randomize a number, then keep shifting by 1 until you hit a number that wsan't used yet.

**Review by Almantask — APPROVED** [review]
I think overall you did a good job. Well done 😄

File: Src/BootCamp.Chapter/LINQExtensions.cs (hunk 2 above)
Comment (matthew-hole) [reply Src/BootCamp.Chapter/LINQExtensions.cs:15]: Ah okay, see I initially did that but my thinking was that for a large dataset that method would be inefficient, I guess I was wrong haha. Thanks for the tip

**Pull request closed without merging**